test/compress/switch.js

```diff
@@ -2447,6 +2447,264 @@ collapse_same_branches_2: {
     expect_stdout: ["PASS", "PASS"]
 }
 
+collapse_same_branches_not_in_a_row: {
+    options = {
+        switches: true,
+        dead_code: true
+    }
+    input: {
+        switch (id(1)) {
+            case 1:
+                console.log("PASS")
+                break;
+            case 2:
+                console.log("FAIL");
+                break;
+            case 3:
+                console.log("PASS");
+                break;
+        }
+    }
+    expect: {
+        switch (id(1)) {
+            case 1:
+            case 3:
+                console.log("PASS");
+                break;
+            case 2:
+                console.log("FAIL");
+        }
+    }
+    expect_stdout: ["PASS"]
+}
+
+collapse_same_branches_not_in_a_row2: {
+    options = {
+        switches: true,
+        dead_code: true
+    }
+    input: {
+        switch (id(1)) {
+            case 1:
+                console.log("PASS");
+                break;
+            case 2:
+                console.log("FAIL");
+                break;
+            case 3:
+                console.log("PASS");
+                break;
+            case 4:
+                console.log("PASS");
+                break;
+        }
+    }
+    expect: {
+        switch (id(1)) {
+            case 1:
+            case 3:
+            case 4:
+                console.log("PASS");
+                break;
+            case 2:
+                console.log("FAIL");
+        }
+    }
+    expect_stdout: ["PASS"]
+}
+
+collapse_same_branches_not_in_a_row_including_fallthrough_with_same_body: {
+    options = {
+        switches: true,
+        dead_code: true
+    }
+    input: {
+        switch (id(1)) {
+            case 1:
+                console.log("PASS");
+                break;
+            case 2:
+                console.log("FAIL");
+                break;
+            case 3:
+            case 4:
+                console.log("PASS");
+                break;
+            case 9:
+                console.log("FAIL");
+                break;
+            case 5:
+            case 6:
+            case 7:
+            case 8:
+                console.log("PASS");
+                break;
+        }
+    }
+    expect: {
+        switch (id(1)) {
+            case 1:
+            case 3:
+            case 4:
+            case 5:
+            case 6:
+            case 7:
+            case 8:
+                console.log("PASS");
+                break;
+            case 2:
+            case 9:
+                console.log("FAIL");
+        }
+    }
+    expect_stdout: ["PASS"]
+}
+
+collapse_same_branches_not_in_a_row_ensure_no_side_effects: {
+    options = {
+        switches: true,
+        dead_code: true
+    }
+    input: {
+        let i = 1;
+        switch (id(2)) {
+            case 1:
+                console.log(1);
+                break;
+            case 2:
+                console.log(2);
+                break;
+            case ++i:
+                console.log(1);
```

Comment:
This test is being skipped because the final branch didn't abort.
```suggestion
                console.log(1);
                break;
```

Reply:
@jridgewell the whole point of this test is to ensure we are **not applying the optimization** in case the switch block has side-effects - so it doesn't really matter in that case.

Reply:
I disagree, we're trying to test that this doesn't apply the optimization _if there's a side-effect_. If the test passes for any other reason (eg, because there's not an abort), then it's not really testing what we want it to.

Reply:
@jridgewell agreed, fixed on 43fe7e79